examples/apigateway/main.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	stdlog "log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/go-kit/kit/endpoint"
+	addsvc "github.com/go-kit/kit/examples/addsvc/client/grpc"
+	"github.com/go-kit/kit/examples/addsvc/server"
+	"github.com/go-kit/kit/loadbalancer"
+	"github.com/go-kit/kit/loadbalancer/consul"
+	log "github.com/go-kit/kit/log"
+	//grpctransport "github.com/go-kit/kit/transport/grpc"
+	httptransport "github.com/go-kit/kit/transport/http"
+	//proto "github.com/golang/protobuf/proto"
+	"github.com/gorilla/mux"
+	"github.com/hashicorp/consul/api"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+func main() {
+	fs := flag.NewFlagSet("", flag.ExitOnError)
+	var (
+		httpAddr   = fs.String("http.addr", ":8000", "Address for HTTP (JSON) server")
+		consulAddr = fs.String("consul.addr", "", "Consul agent address")
+	)
+	flag.Usage = fs.Usage
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		fmt.Fprintf(os.Stderr, "%v", err)
+		os.Exit(1)
+	}
+
+	// log
+	logger := log.NewLogfmtLogger(os.Stderr)
+	logger = log.NewContext(logger).With("ts", log.DefaultTimestampUTC).With("caller", log.DefaultCaller)
+	stdlog.SetFlags(0)                             // flags are handled by Go kit's logger
+	stdlog.SetOutput(log.NewStdlibAdapter(logger)) // redirect anything using stdlib log to us
+
+	// errors
+	errc := make(chan error)
```

Comment:
Please put this down with the other goroutine pushing to errc. That would help reveal a bug: this channel needs a buffer size of 2 :)